src/components/panel/panel.js

```diff
@@ -1563,36 +1636,161 @@ MdPanelAnimation.prototype.closeTo = function(closeTo) {
  * Returns the element and bounds for the animation target.
  * @param {string|!Element|{top: number, left: number}} location
  * @returns {!{element: !angular.JQLite|undefined, bounds: !DOMRect}}
+ * @private
  */
-MdPanelAnimation.prototype.getPanelAnimationTarget = function(location) {
+MdPanelAnimation.prototype._getPanelAnimationTarget = function(location) {
   if (angular.isDefined(location.top) || angular.isDefined(location.left)) {
     return {
       element: undefined,
       bounds: {
         top: location.top || 0,
-        left: location.left || 0,
-        height: 1,
-        width: 1
+        left: location.left || 0
       }
     };
   } else {
-    return this._getBoundingClientRect(this._getElement(location));
+    return this._getBoundingClientRect(getElement(location));
   }
-}
+};
 
 
 /**
- * TODO(KarenParker): Switch and update the shared version of this method that
- * others are currently building.
- * Gets the element from the string|element.
- * @param {string|!Element} el
- * @returns {!angular.JQLite}
+ * Specifies the animation class.
+ *
+ * There are several default animations that can be used:
+ * (MdPanelAnimation.animation)
+ *   SLIDE: The panel slides in and out from the specified
+ *        elements.
+ *   SCALE: The panel scales in and out.
+ *   FADE: The panel fades in and out.
+ *
+ * @param {string|!{open: string, close: string}} cssClass
+ * @returns {MdPanelAnimation}
+ */
+
+MdPanelAnimation.prototype.withAnimation = function(cssClass) {
+  this._animationClass = cssClass;
+  return this;
+};
+
+
+/**
+ * Animate the panel open.
+ * @param {!angular.JQLite} panelEl
+ * @param animator
+ * @returns {!angular.$q.Promise}
+ */
+MdPanelAnimation.prototype.animateOpen = function(panelEl, animator) {
+  this._fixBounds(panelEl);
```

Comment:
optional: this function is pretty long and could benefit from being broken into smaller functions